From df8825fb7bdfb3af108a89822b50b62fc3ebefe7 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Niklas=20S=C3=B6derlund?=
 <niklas.soderlund+renesas@ragnatech.se>
Date: Mon, 15 Sep 2025 19:17:16 +0200
Subject: [PATCH 3/3] media: rppx1: Backport upstream bug fixes
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Backport to BSP fixes that have been squashed in the correct commits for
upstreaming.

  - When the AWEG are applied.
  - Add missing error check for module processing.
  - Add a missing header include.

Signed-off-by: Niklas Söderlund <niklas.soderlund+renesas@ragnatech.se>
---
 drivers/media/platform/dreamchip/rppx1/rpp_params.c | 9 +++++++--
 drivers/media/platform/dreamchip/rppx1/rppx1.c      | 1 +
 2 files changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/media/platform/dreamchip/rppx1/rpp_params.c b/drivers/media/platform/dreamchip/rppx1/rpp_params.c
index d40105e4f782..b7b8adfe9337 100644
--- a/drivers/media/platform/dreamchip/rppx1/rpp_params.c
+++ b/drivers/media/platform/dreamchip/rppx1/rpp_params.c
@@ -19,6 +19,7 @@ int rppx1_params_rkisp1(struct rppx1 *rpp, struct rkisp1_ext_params_cfg *cfg,
 		const union rppx1_params_rkisp1_config *block =
 			(const union rppx1_params_rkisp1_config *)&cfg->data[block_offset];
 		struct rpp_module *module;
+		int ret;
 
 		block_offset += block->header.size;
 
@@ -27,7 +28,7 @@ int rppx1_params_rkisp1(struct rppx1 *rpp, struct rkisp1_ext_params_cfg *cfg,
 			module = &rpp->pre1.bls;
 			break;
 		case RKISP1_EXT_PARAMS_BLOCK_TYPE_AWB_GAIN:
-			module = &rpp->post.awbg;
+			module = &rpp->pre1.awbg;
 			break;
 		case RKISP1_EXT_PARAMS_BLOCK_TYPE_FLT:
 		case RKISP1_EXT_PARAMS_BLOCK_TYPE_BDM:
@@ -67,7 +68,11 @@ int rppx1_params_rkisp1(struct rppx1 *rpp, struct rkisp1_ext_params_cfg *cfg,
 			continue;
 		}
 
-		rpp_module_call(module, param_rkisp1, block, write, priv);
+		ret = rpp_module_call(module, param_rkisp1, block, write, priv);
+		if (ret) {
+			pr_err("Error processing RPPX1 block type: 0x%04x\n", block->header.type);
+			return ret;
+		}
 	}
 
 	return 0;
diff --git a/drivers/media/platform/dreamchip/rppx1/rppx1.c b/drivers/media/platform/dreamchip/rppx1/rppx1.c
index c2a0deb4d253..a86244a86ab8 100644
--- a/drivers/media/platform/dreamchip/rppx1/rppx1.c
+++ b/drivers/media/platform/dreamchip/rppx1/rppx1.c
@@ -9,6 +9,7 @@
 
 #include <linux/io.h>
 #include <linux/module.h>
+#include <linux/slab.h>
 
 #include "rppx1.h"
 
-- 
2.25.1

